src/sys/resource.rs

```diff
@@ -0,0 +1,49 @@
+use std::mem;
+
+use libc::{self, c_int};
+pub use libc::{rlimit, RLIM_INFINITY};
+
+use {Errno, Result};
+
+#[repr(i32)]
```

Comment:
Using the type `i32` here is dangerous. Who says that the constants are represented by 32bit integers on all (current and future) platforms?

I got around that in my pull request by using the default (`isize`) and casting the constants to `isize` in the enumeration, for example: `RLIMIT_CORE = libc::RLIMIT_CORE as isize`.

I would prefer to write `#[repr(libc::c_int)]` or whatever type is used for these constants at the moment in libc. But even that can differ from platform to platform.

Is there a way to conditionalize the repr line on platforms? We certainly can conditionalise the whole enumeration definition on the platform, but that is a lot of repetition, though that certainly can be dealt with using macros.